glib/SafeObjectHandle.cs

```diff
@@ -0,0 +1,113 @@
+﻿// SafeObjectHandle.cs - SafeHandle implementation for GObject
+//
+// Authors: Marius Ungureanu <maungu@microsoft.com>
+//
+// Copyright (c) 2017-2017 Microsoft, Inc.
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of version 2 of the Lesser GNU General
+// Public License as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this program; if not, write to the
+// Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+// Boston, MA 02111-1307, USA.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Win32.SafeHandles;
+
+namespace GLib
+{
+	public class SafeObjectHandle : SafeHandleZeroOrMinusOneIsInvalid
```

Comment:
Public for a reason. Subclassing so we can replace the safehandle and still have toggleref support